model/workflow.go

```diff
@@ -176,6 +178,17 @@ func (w *Workflow) UnmarshalJSON(data []byte) error {
 			if nestedData, err = getBytesFromFile(s); err != nil {
 				return err
 			}
+
+			// TODO: optimize this
+			// In the specification, we can declared independently definitions with another file format, so
+			// we must convert the independently yaml source to json before unmarshal.
```

Comment:
@lsytj0413 wdyt about opening an issue for this one?

Reply:
nice advice, i will open an issue and fix it in another PR.